File: Data Structures & Algorithms/merge-two-sorted-linked-lists/submission-1.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
		if (list1 == null) {
				return list2;
			}
			if (list2 == null) {
				return list1;
			}
        ListNode newHead = (list1.val<=list2.val)?list1:list2;

        ListNode curr1 = list1;
        ListNode curr2 = list2;
        ListNode prev1 = new ListNode(0,null);

        while (curr1 != null && curr2 != null) {
            if (curr1.val <= curr2.val) {
                prev1 = curr1;
                curr1 = curr1.next;
                continue;
            }
            ListNode temp = curr2;
            curr2 = curr2.next;

            prev1.next = temp;
            temp.next = curr1;
            prev1 = temp;

        }
        if (curr2!=null) {
            prev1.next = curr2;
        }
        return newHead;
        
    }
}

File: Data Structures & Algorithms/insert-greatest-common-divisors-in-linked-list/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def insertGreatestCommonDivisors(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head.next:
            p1 = head
            p2 = head.next
            while p2:
                res = 0
               
                res= math.gcd(p1.val, p2.val)

                newNode = ListNode(res)
                p1.next = newNode
                newNode.next = p2
                p1 = p1.next.next
                p2 = p2.next
        return head
  

        